sound/hda/intel-sdw-acpi.c

```diff
@@ -58,19 +58,20 @@ sdw_intel_scan_controller(struct sdw_intel_acpi_info *info)
 {
 	struct acpi_device *adev = acpi_fetch_acpi_dev(info->handle);
 	struct fwnode_handle *fwnode;
+	unsigned long list_ul;
```

Comment:
I believe the same comment applies here, i.e. splitting out the helper that will return `list` in `unsigned long` form. This will make code cleaner. But it might be that you do in the sequential patches (haven't seen them yet).

Reply:
I meant that instead of adding a new variable you can make a helper that will return an unsigned long for all cases and use bit ops on top of that. When I come to work after vacations (mid-August) I might be able to draft it.

Reply:
ok, I won't have the time to work on this so will wait for your suggestion @andy-shev.

Reply:
> ok, I won't have the time to work on this so will wait for your suggestion @andy-shev.

Sure, just ping me then.

Reply:
Personally I think the problem really is the nest of gotos, not sure a helper function is strictly necessary, although fine by me. I would probably suggest just drop the inclusion of both strings in the error message, which I assume is driving the gotos. I would probably also move the conversion to unsigned long up to the parsing as well something like:

        unsigned long list;
        u32 count, tmp;

        ret = fwnode_property_read_u32(fwnode, "mipi-sdw-manager-list", &tmp);
        if (ret) {
                ret = fwnode_property_read_u32(fwnode, "mipi-sdw-master-count", &count);
                if (ret) {
                        dev_err(&adev->dev, "Failed to read mipi-sdw-manager-list: %d\n", ret);
                        return ret;
                }

                list = GENMASK(count - 1, 0);
        } else {
                list = tmp;
                count = hweight32(list);
        }

Rest of this series looks good to me though.

Reply:
> Personally I think the problem really is the nest of gotos, not sure a helper function is strictly necessary, although fine by me. I would probably suggest just drop the inclusion of both strings in the error message, which I assume is driving the gotos. I would probably also move the conversion to unsigned long up to the parsing as well something like: ...

LGTM.

Reply:
changed the code as suggested, thanks for the feedback @andy-shev @charleskeepax